documentation/tuning_guide.markdown

```diff
@@ -47,113 +47,103 @@ setting.  The value of this setting is used by Puppet Server to determine how
+to the pool. At that point, the next thread can "borrow" the JRuby instance to
+use to handle the requests from the second agent.
 
 Assuming you have more than one CPU core in your machine, this situation means
 that you won't be getting the maximum possible throughput from your Puppet Server
-installation.  Increasing the value from `1` to `2` would mean that Puppet Server
+installation. Increasing the value from `1` to `2` would mean that Puppet Server
 could now use a second CPU core to handle the requests from a second Puppet agent
 simultaneously.
 
 It follows, then, that the maximum sensible value to use for this setting will
-be roughly the number of CPU cores you have in your server.  Setting the value
+be roughly the number of CPU cores you have in your server. Setting the value
 to something much higher than that won't improve performance, because even if there
 are extra JRuby instances available in the pool to do work, they won't be able
 to actually do any work if all of the CPU cores are already busy using JRuby
 instances to handle incoming agent requests.
 
-(There are exceptions to this rule; if you have report processors that make
-a network connection as part of the processing of a report, and if there is a chance
+(There are some exceptions to this rule. For example, if you have report processors that make a network connection as part of the processing of a report, and if there is a chance
 that the network operation is slow and will block on I/O for some period of time,
-then it may make sense to have more JRuby instances than the number of cores,
-because the JVM will be smart enough to suspend the thread that is handling those
-kinds of requests and use the CPUs for other work, assuming there are still JRuby
-instances available in the pool.  In a case like this you might want to set
-`max-active-instances` to a value higher than the number of CPUs.)
-
-At this point you may be wondering "what's the downside to just setting
-`max-active-instances` to a really high value?"  The answer to this question, in
-a nutshell, is "memory usage".  This brings us to the other extremely important
+then it might make sense to have more JRuby instances than the number of cores. The JVM is smart enough to suspend the thread that is handling those kinds of requests and use the CPUs for other work, assuming there are still JRuby instances available in the pool. In a case like this you might want to set `max-active-instances` to a value higher than the number of CPUs.)
+
+At this point you may be wondering, "What's the downside to just setting
+`max-active-instances` to a really high value?" The answer to this question, in
+a nutshell, is "memory usage". This brings us to the other extremely important
```

Comment:
The last sentence in the paragraph reads a little strange to me here.  Maybe should say something like, "This brings us to the other extremely important *setting to consider* for Puppet Server?"